src/vm/diagnosticserver.cpp

```diff
@@ -46,34 +47,32 @@ static DWORD WINAPI DiagnosticsServerThread(LPVOID lpThreadParameter)
             if (pStream == nullptr)
                 continue;
 
-            // TODO: Read operation should happen in a loop.
-            uint32_t nNumberOfBytesRead = 0;
-            MessageHeader header;
-            bool fSuccess = pStream->Read(&header, sizeof(header), nNumberOfBytesRead);
-            if (!fSuccess || nNumberOfBytesRead != sizeof(header))
+            DiagnosticsIpc::IpcMessage message(pStream);
+
+            if (!::strcmp((char *)message.GetHeader().Magic, DOTNET_IPC_V1_MAGIC))
```

Comment:
strcmp is supposed to return 0 when the strings are equal, and you are failing the connection on success?

Reply:
Yup!  Caught that one :)